lib/morpheus.gi

```diff
@@ -2818,6 +2818,44 @@ local d,iso,a,b,c,o,s,two,rt,r,z,e,y,re,m,gens,cnt,lim,p,
 end);
 
 
+BindGlobal("IsomorphismPGroups",function(G,H)
+local s,p,eG,eH,fG,fH,pc,imgs;
+  s:=Size(G);
+  if Size(H)<>s then return fail;fi;
+  p:=Collected(Factors(s));
+  if Length(p)>1 then TryNextMethod();fi;
+  p:=p[1][1];
+  eG:=CallFuncList(ValueGlobal("EpimorphismPqStandardPresentation"),[G]:
+    Prime:=p);
+  eH:=CallFuncList(ValueGlobal("EpimorphismPqStandardPresentation"),[H]:
+    Prime:=p);
+
+  # check presentations
+  fG:=Range(eG);
+  fH:=Range(eH);
+  if List(RelatorsOfFpGroup(fG),
+    x->MappedWord(x,FreeGeneratorsOfFpGroup(fG),FreeGeneratorsOfFpGroup(fH)))
+      <>RelatorsOfFpGroup(fH) then
+    return fail;
+  fi;
+
+  # move to pc pres
+  pc:=PcGroupFpGroup(fG);
+
+  # new maps to pc
+  imgs:=List(MappingGeneratorsImages(eG)[2],
+    x->MappedWord(UnderlyingElement(x),
+      FreeGeneratorsOfFpGroup(fG),GeneratorsOfGroup(pc)));
+  eG:=GroupHomomorphismByImages(G,pc,MappingGeneratorsImages(eG)[1],imgs);
```

Comment:
Maybe use the `NC` version here and below? Or do you want the extra checks?
```suggestion
  eG:=GroupHomomorphismByImagesNC(G,pc,MappingGeneratorsImages(eG)[1],imgs);
```

Reply:
Since an error (maybe in how I interpret the PQ results) would be disastrous, and the overall calculation is costly, the test here did seem to be worth doing, thus deliberately no NC.

Reply:
Let me just add, that indeed this already caught a problem (which is fixed in a second commit)